### ISSUE TITLE
Decoding error while using the model transfer from pretrained model

### ISSUE THREAD

**Issue by Subohao** [espnet/espnet#4302]
**Describe the bug**
Dear developers, 
I met some problems while applying the fine-tuning asr model to decode the test set, and I found no further information on solving this bug. Thanks for helping me!!
I think the bug is about the BeamSearchTransducer function initialization of asr_inference.py, and the description is provided below:

```
# python3 -m espnet2.bin.asr_inference --batch_size 1 --ngpu 1 --data_path_and_name_and_type dump/raw/test1/wav.scp,speech,sound --key_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/keys.1.scp --asr_train_config exp/asr_transfer_learning_with_pre_trained_model_from_HF/config.yaml --asr_model_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/valid.loss.ave_10best.pth --output_dir exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/output.1 --lm_train_config exp/lm_train_bpe38/config.yaml --lm_file exp/lm_train_bpe38/valid.loss.ave.pth 
# Started at Tue Apr 19 17:54:49 CST 2022
#
/home/subohao/.conda/envs/espnet_py38/bin/python3 /opt/software/espnet/espnet2/bin/asr_inference.py --batch_size 1 --ngpu 1 --data_path_and_name_and_type dump/raw/test1/wav.scp,speech,sound --key_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/keys.1.scp --asr_train_config exp/asr_transfer_learning_with_pre_trained_model_from_HF/config.yaml --asr_model_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/valid.loss.ave_10best.pth --output_dir exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/output.1 --lm_train_config exp/lm_train_bpe38/config.yaml --lm_file exp/lm_train_bpe38/valid.loss.ave.pth
2022-04-19 17:54:51,931 (asr:411) INFO: Vocabulary size: 38
2022-04-19 17:54:52,060 (conformer_encoder:137) WARNING: Using legacy_rel_pos and it will be deprecated in the future.
2022-04-19 17:54:52,119 (conformer_encoder:237) WARNING: Using legacy_rel_selfattn and it will be deprecated in the future.
2022-04-19 17:54:57,923 (lm:199) INFO: Vocabulary size: 38
Traceback (most recent call last):
  File "/home/subohao/.conda/envs/espnet_py38/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/subohao/.conda/envs/espnet_py38/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 609, in <module>
    main()
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 605, in main
    inference(**kwargs)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 395, in inference
    speech2text = Speech2Text.from_pretrained(
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 317, in from_pretrained
    return Speech2Text(**kwargs)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 123, in __init__
    beam_search_transducer = BeamSearchTransducer(
TypeError: type object argument after ** must be a mapping, not NoneType
# Accounting: time=9 threads=1
# Ended (code 1) at Tue Apr 19 17:54:58 CST 2022, elapsed time 9 seconds
```

**Basic environments:**
 - OS information: Linux 5.4.0-105-generic #119-Ubuntu SMP Mon Mar 7 18:49:24 UTC 2022 x86_64
 - python version: `3.8.0 (default, Nov  6 2019, 21:49:08)  [GCC 7.3.0]
 - espnet version: `espnet 0.10.7a1`
 - Git hash: `189e1593dcd5e4c74b84e9a04a6da3906f64b40f`
   - Commit date `Wed Apr 6 16:42:01 2022 -0400`
 - pytorch version: `pytorch 1.10.0`

**Environments from `torch.utils.collect_env`:**

```
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.0 (default, Nov  6 2019, 21:49:08)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-105-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.0.221
GPU models and configuration: 
GPU 0: A100-SXM-80GB
GPU 1: A100-SXM-80GB
GPU 2: A100-SXM-80GB
GPU 3: DGX Display
GPU 4: A100-SXM-80GB

Nvidia driver version: 450.172.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] numpydoc==1.2
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.10.0
[pip3] torch-complex==0.4.3
[pip3] torch-optimizer==0.3.0
[pip3] torchaudio==0.10.0
[pip3] warprnnt-pytorch==0.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.1.74              h6bb024c_0    nvidia
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.1            py38hd3c417c_0  
[conda] mkl_random                1.2.2            py38h51133e4_0  
[conda] mypy_extensions           0.4.3            py38h06a4308_1  
[conda] numpy                     1.21.2           py38h20f2e39_0  
[conda] numpy-base                1.21.2           py38h79a1101_0  
[conda] numpydoc                  1.2                pyhd3eb1b0_0  
[conda] pytorch                   1.10.0          py3.8_cuda11.1_cudnn8.0.5_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] pytorch-ranger            0.1.1                    pypi_0    pypi
[conda] pytorch-wpe               0.0.1                    pypi_0    pypi
[conda] torch-complex             0.4.3                    pypi_0    pypi
[conda] torch-optimizer           0.3.0                    pypi_0    pypi
[conda] torchaudio                0.10.0               py38_cu111    pytorch
[conda] warprnnt-pytorch          0.1                       dev_0    <develop>
```

**Task information:**
 - Task: ASR
 - Recipe: TEMPLATE/asr1/asr.sh
 - ESPnet2

**To Reproduce**
1. Download the pre-trained model (chai_librispeech_asr_train_rnnt_conformer_raw_en_bpe5000_sp-15epoch.pth) from [Huggingface](https://huggingface.co/espnet/chai_librispeech_asr_train_rnnt_conformer_raw_en_bpe5000_sp/tree/main/exp/asr_train_rnnt_conformer_ngpu4_raw_en_bpe5000_sp)
2. transfer learning with my own dataset following the tutorial in (document)[https://espnet.github.io/espnet/notebook/espnet2_asr_transfer_learning_demo.html#mini_an4-recipe-as-a-transfer-learning-example]
3. I get the model after finetuning the pretrained model, then I use the command : `./asr.sh --stage 12 --stop_stage 12 --train-set "train1" --valid-set "dev1" --test_sets "test1" --asr_config "conf/train_rnnt_conformer.yaml" --asr_tag "transfer_learning_with_pre_trained_model_from_HF" --pretrained_model "15epoch.pth" --ignore_init_mismatch "true" --inference_asr_model valid.loss.ave_10best.pth` to decode the test data.
4. Then the error shows as the following description

**Error logs**
```
# python3 -m espnet2.bin.asr_inference --batch_size 1 --ngpu 1 --data_path_and_name_and_type dump/raw/test1/wav.scp,speech,sound --key_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/keys.1.scp --asr_train_config exp/asr_transfer_learning_with_pre_trained_model_from_HF/config.yaml --asr_model_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/valid.loss.ave_10best.pth --output_dir exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/output.1 --lm_train_config exp/lm_train_bpe38/config.yaml --lm_file exp/lm_train_bpe38/valid.loss.ave.pth 
# Started at Tue Apr 19 17:54:49 CST 2022
#
/home/subohao/.conda/envs/espnet_py38/bin/python3 /opt/software/espnet/espnet2/bin/asr_inference.py --batch_size 1 --ngpu 1 --data_path_and_name_and_type dump/raw/test1/wav.scp,speech,sound --key_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/keys.1.scp --asr_train_config exp/asr_transfer_learning_with_pre_trained_model_from_HF/config.yaml --asr_model_file exp/asr_transfer_learning_with_pre_trained_model_from_HF/valid.loss.ave_10best.pth --output_dir exp/asr_transfer_learning_with_pre_trained_model_from_HF/inference_lm_lm_train_bpe38_valid.loss.ave_asr_model_valid.loss.ave_10best/test1/logdir/output.1 --lm_train_config exp/lm_train_bpe38/config.yaml --lm_file exp/lm_train_bpe38/valid.loss.ave.pth
2022-04-19 17:54:51,931 (asr:411) INFO: Vocabulary size: 38
2022-04-19 17:54:52,060 (conformer_encoder:137) WARNING: Using legacy_rel_pos and it will be deprecated in the future.
2022-04-19 17:54:52,119 (conformer_encoder:237) WARNING: Using legacy_rel_selfattn and it will be deprecated in the future.
2022-04-19 17:54:57,923 (lm:199) INFO: Vocabulary size: 38
Traceback (most recent call last):
  File "/home/subohao/.conda/envs/espnet_py38/lib/python3.8/runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/subohao/.conda/envs/espnet_py38/lib/python3.8/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 609, in <module>
    main()
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 605, in main
    inference(**kwargs)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 395, in inference
    speech2text = Speech2Text.from_pretrained(
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 317, in from_pretrained
    return Speech2Text(**kwargs)
  File "/opt/software/espnet/espnet2/bin/asr_inference.py", line 123, in __init__
    beam_search_transducer = BeamSearchTransducer(
TypeError: type object argument after ** must be a mapping, not NoneType
# Accounting: time=9 threads=1
# Ended (code 1) at Tue Apr 19 17:54:58 CST 2022, elapsed time 9 seconds
```

**Comment by b-flo** [comment]
Hi,

`transducer_conf` is None which causes the error [here](https://github.com/espnet/espnet/blob/master/espnet2/bin/asr_inference.py#L145=). Until we fix it, I suppose you can either comment the line or manually set [these parameters](https://github.com/espnet/espnet/blob/master/espnet2/asr/transducer/beam_search_transducer.py#L54-L63=) to avoid `TypeError`.

P.S: It seems your repo is not up-to-date given the error lines. It's not an issue here but be careful with the link I gave, the line numbers I point out are not the same.

**Comment by Subohao** [comment]
@b-flo oh, thanks for the quick response.
I have commented on the `**transducer_conf` and directly set parameters.
It seems working now, really appreciate that!

**Comment by b-flo** [comment]
For reference: it was fixed in #4032. I'm keeping this issue open until the PR is merged or the bug is fixed in another PR.
